diff --git a/common/os_calls.c b/common/os_calls.c
index 76685a40..b9d32edc 100644
--- a/common/os_calls.c
+++ b/common/os_calls.c
@@ -122,7 +122,7 @@ g_mk_temp_dir(const char *app_name)
                 return 1;
             }
         }
-        g_chmod_hex("/tmp/.xrdp", 0x1777);
+        g_chmod_hex("/tmp/.xrdp", 0x3777);
     }
     return 0;
 }
diff --git a/xorg/X11R7.6/rdp/rdpup.c b/xorg/X11R7.6/rdp/rdpup.c
index 42435f8b..ff36bd62 100644
--- a/xorg/X11R7.6/rdp/rdpup.c
+++ b/xorg/X11R7.6/rdp/rdpup.c
@@ -1242,7 +1242,7 @@ rdpup_init(void)
             return 0;
         }

-        g_chmod_hex("/tmp/.xrdp", 0x1777);
+        g_chmod_hex("/tmp/.xrdp", 0x3777);
     }

     i = atoi(display);
diff --git a/xorgxrdp/module/rdpClientCon.c b/xorgxrdp/module/rdpClientCon.c
index 5bcb923..aaec31c 100644
--- a/xorgxrdp/module/rdpClientCon.c
+++ b/xorgxrdp/module/rdpClientCon.c
@@ -1184,7 +1184,7 @@ rdpClientConInit(rdpPtr dev)
                 return 0;
             }
         }
-        g_chmod_hex("/tmp/.xrdp", 0x1777);
+        g_chmod_hex("/tmp/.xrdp", 0x3777);
     }
     i = atoi(display);
     if (i < 1)
@@ -1203,6 +1203,7 @@ rdpClientConInit(rdpPtr dev)
             return 1;
         }
         g_sck_listen(dev->listen_sck);
+        g_chmod_hex(dev->uds_data, 0x0660);
         rdpClientConAddEnabledDevice(dev->pScreen, dev->listen_sck);
     }

