diff --git a/instfiles/xrdp-sesman.service b/instfiles/xrdp-sesman.service
index ff8d5b0c..c92e7ea7 100644
--- a/instfiles/xrdp-sesman.service
+++ b/instfiles/xrdp-sesman.service
@@ -7,7 +7,8 @@ BindsTo=xrdp.service

 [Service]
 Type=forking
-PIDFile=/var/run/xrdp-sesman.pid
+PIDFile=/run/xrdp/xrdp-sesman.pid
+RuntimeDirectory=xrdp
 EnvironmentFile=-/etc/sysconfig/xrdp
 EnvironmentFile=-/etc/default/xrdp
 ExecStart=/usr/sbin/xrdp-sesman $SESMAN_OPTIONS
diff --git a/instfiles/xrdp.service b/instfiles/xrdp.service
index 590502fb..d3575b88 100644
--- a/instfiles/xrdp.service
+++ b/instfiles/xrdp.service
@@ -6,10 +6,13 @@ After=network.target xrdp-sesman.service

 [Service]
 Type=forking
-PIDFile=/var/run/xrdp.pid
+PIDFile=/run/xrdp/xrdp.pid
+RuntimeDirectory=xrdp
 EnvironmentFile=-/etc/sysconfig/xrdp
 EnvironmentFile=-/etc/default/xrdp
 PermissionsStartOnly=true
+User=xrdp
+Group=xrdp
 ExecStartPre=/bin/sh /usr/share/xrdp/socksetup
 ExecStart=/usr/sbin/xrdp $XRDP_OPTIONS
 ExecStop=/usr/sbin/xrdp $XRDP_OPTIONS --kill
